Fix/rerunning cukes

ulferts opened the pull request:
Consists of two parts:

1) fix rerunning failed cukes: The failed cukes are rerun automatically by travis. However, the step definitions where not included so that cukes that are rerun pass because of that.

2) Adapt cukes that got broken while problem 1) existed.

3) fix behaviour that got broken without anybody noticing: https://community.openproject.org/work_packages/22451/activity

ulferts committed:
include step definitions when rerunning failed cukes

ulferts committed:
adapt cukes to changed wiki menu entry

ulferts committed:
adapt cuke to be able to add a comment

ulferts committed:
fix seeing all work package attributes

I am not sure why click_link_or_button is not working here

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://ulferts-fix-rerunning-cukes.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://ulferts-fix-rerunning-cukes.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

ulferts committed:
keep values on work package type switch

the former solution based on introducing a custom attribute does not work because the allowed values are updated when the type is switched but it is updated without the custom attribute being recalculated. The selected value is thus lost and an invalid field (empty option prepended) is selected instead

ulferts commented:
https://community.openproject.org/work_packages/22451/activity

oliverguenther commented:
Good catch. It's quite scary that nobody has seen this since this landed on dev. The changes LGTM.
